cicecore/cicedynB/dynamics/ice_dyn_shared.F90

```diff
@@ -1276,8 +1279,100 @@ subroutine deformations (nx_block,   ny_block,   &
 
       enddo                     ! ij
 
-      end subroutine deformations
+    end subroutine deformations
+    
+!=======================================================================
+
+! Compute deformations for mechanical redistribution at T point
+!
+! author: JF Lemieux, ECCC
+! Nov 2021
+
+      subroutine deformations_T (nx_block,   ny_block,   &
+                                 icellt,                 &
+                                 indxti,     indxtj,     &
+                                 uvelE,      vvelE,      &
+                                 uvelN,      vvelN,      &
+                                 dxN,        dyE,        &
+                                 dxT,        dyT,        &
+                                 tarear,                 &
+                                 shear,      divu,       &
+                                 rdg_conv,   rdg_shear )
```

Comment:
This whole subroutine looks like it's the same as `deformations`, only calling  `strain_ratesT` instead of `strain_rates`.

Should we instead keep the existing `deformations` subroutine, and add an `if (grid_system == 'B') .. else ..` block for calling `strain_rates` vs  `strain_rates` ?

Reply:
Good eyes, @phil-blain .
I would prefer to reuse as much code as possible.  Initially it's fine to have duplicative code, while developing, but if it makes sense and is easy to merge/clean up enroute, let's do it...

Reply:
Yeah, upon further inspection with @JFLemieux73, it's not exactly the same, as the B grid version does the average of the 4 corners: 
https://github.com/CICE-Consortium/CICE/blob/0f75f068d63725062da4ea944443fbdd6dbed2fb/cicecore/cicedynB/dynamics/ice_dyn_shared.F90#L1269-L1278

whereas the CD-grid version already has the quantities at the tracer point, so it just computes the deformations without averaging:
https://github.com/CICE-Consortium/CICE/blob/0f75f068d63725062da4ea944443fbdd6dbed2fb/cicecore/cicedynB/dynamics/ice_dyn_shared.F90#L1363-L1370